### PR TITLE
Provide option to catch or throw exception in FeatureTransformer

### PR DESCRIPTION
**Pull request by SeaOfOcean**
## What changes were proposed in this pull request?
Provide option to catch or throw exception in FeatureTransformer

## How was this patch tested?
unit test

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/2228

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/FeatureTransformer.scala [hunk 1]
```diff
@@ -23,8 +23,11 @@ import org.apache.log4j.Logger
 
 /**
  * FeatureTransformer is a transformer that transform ImageFeature
+ * @param catchException if true, catch the exception of the transformer to avoid crashing.
+ *                       if false, interrupt the transformer when error happens
  */
-abstract class FeatureTransformer() extends Transformer[ImageFeature, ImageFeature] {
+abstract class FeatureTransformer(catchException: Boolean = FeatureTransformer.catchException)
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/FeatureTransformer.scala]: why set default value to `FeatureTransformer.catchException` ?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/FeatureTransformer.scala [hunk 2]
```diff
@@ -95,6 +106,15 @@ abstract class FeatureTransformer() extends Transformer[ImageFeature, ImageFeatu
   override def -> [C](other: Transformer[ImageFeature, C]): Transformer[ImageFeature, C] = {
     new ChainedTransformer(this, other)
   }
+
+  /**
+   * if true, catch the exception of the transformer to avoid crashing.
+   * if false, interrupt the transformer when error happens
+   */
+  def setIgnoreException(ignore: Boolean = true): this.type = {
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/FeatureTransformer.scala]: enable + disable function is better than a Boolean param

**Comment by SeaOfOcean** [comment]
pr validate pass

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/FeatureTransformer.scala [hunk 3]
```diff
@@ -24,12 +24,19 @@ import org.apache.log4j.Logger
...
 
   import FeatureTransformer.logger
 
   private var outKey: Option[String] = None
 
+  /**
+   * if true, catch the exception of the transformer to avoid crashing.
+   * if false, interrupt the transformer when error happens
+   */
+  var ignoreException: Boolean = false
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/transform/vision/image/FeatureTransformer.scala]: private

**Pull request merged**